src/Microsoft.Azure.SignalR.Common/Utilities/ClaimsUtility.cs

```diff
@@ -168,5 +175,48 @@ internal static ClaimsPrincipal GetUserPrincipal(Claim[] messageClaims)
 
             return new ClaimsPrincipal(new ClaimsIdentity(claims, authenticationType, nameType, roleType));
         }
+
+        private static bool TryGetCustomHandshakeTimeoutClaim(TimeSpan? handshakeTimeout, ILogger logger, out Claim claim)
+        {
+            // use default handshake timeout
+            if (!handshakeTimeout.HasValue || handshakeTimeout.Value.Equals(Constants.Periods.DefaultHandshakeTimeout))
+            {
+                claim = null;
+                return false;
+            }
+
+            // the custom handshake timeout is invalid, use default hanshake timeout instead
+            if (handshakeTimeout.Value.CompareTo(TimeSpan.Zero) <= 0 ||
+                handshakeTimeout.Value.CompareTo(Constants.Periods.MaxCustomHandshakeTimeout) > 0)
+            {
+                Log.FailToSetCustomHandshakeTimeout(logger, new ArgumentOutOfRangeException(nameof(handshakeTimeout)));
```

Comment:
move the check to some validation when startup? otherwise customer gets such log for every request